tests/neg/i1050c.scala

```diff
@@ -4,7 +4,7 @@ object Import {
     trait B { type L >: Any}
     trait U {
       lazy val p: B
-      locally { val x: p.L = ??? } // old-error: nonfinal lazy
+      locally { val x: p.L = ??? } // error: nonfinal lazy
```

Comment:
`p.L` has bad bounds, and this test was disabled (by mistake) by unrelated PR #1106 (probably because it had broken), so re-enabling this test.

Since it isn't about a method call it's not addressed by the current fix, but it's still related to realizability checks I guess.

Reply:
No, it was because the tests in 1050c are done during typer and the non-final lazy is detected later in PostTyper, so not detected when there are typer errors. The comment was misleading.
There are tests like this in i1050.scala, which get reported correctly.